## 86% of end-end tests have passed
```
Total executed tests: 14
Failed tests: 2
Expected to fail tests: 0
Passed tests: 12
```
```
IDs of failed tests: 741839,741841 
```
<h3>Failed tests (2)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=fix-erc20-bridge-quick-settings2&PR_ID=22242&APK_NAME=22242.apk&TR_CASE_IDS=741839,741841">Rerun failed tests</a></li><br/><h4>Class TestWalletCollectibles:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3808005">test_wallet_collectibles_balance</a>, id: 741839 </b></td></tr><tr><td><p><blockquote><div># STEP: Check BVL collectible info and image</div><div># STEP: Check Glitch Punks collectible info and image</div></blockquote></p>

```
critical/wallet/test_collectibles.py:94: in test_wallet_collectibles_balance
    self.errors.verify_no_errors()
base_test_case.py:179: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Device 1: Collectible 'BVL' is not displayed
E    Device 1: Collectible 'Glitch Punks' is not displayed
```

<br/><br/></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3808007">test_wallet_collectible_send_from_expanded_info_view</a>, id: 741841 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `accessibility id`: `collectibles-tab` </div></blockquote></p>

```
critical/wallet/test_collectibles.py:149: in test_wallet_collectible_send_from_expanded_info_view
    self.wallet_view.collectibles_tab.click()
../views/base_element.py:89: in click
    element = self.find_element()
../views/base_element.py:78: in find_element
    raise NoSuchElementException(
 Device 1: Button by accessibility id: `collectibles-tab` is not found on the screen; For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
```

<br/><br/></td></tr></tbody></table></details><h3>Passed tests (12)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestWalletCollectibles:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3808006">test_wallet_send_collectible</a>, id: 741840 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3807996">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESdPJvmSXNHYP2wkiE">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESRRRa7TMwGHyh3U9t">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3808003">test_wallet_swap_flow_mainnet</a>, id: 741555 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3808000">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3808001">test_wallet_balance_mainnet</a>, id: 740490 </b></td></tr><tr><td></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3808004">test_wallet_bridge_flow_mainnet</a>, id: 741612 </b></td></tr><tr><td></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3808002">test_wallet_send_flow_mainnet</a>, id: 741554 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3807999">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3807998">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3807994">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES9aB1CUrjvgw3ileO">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3807997">test_restore_multiaccount_with_waku_backup_remove_profile_switch</a>, id: 703133 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTES9aB1CUrjvgw3ileO">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3807995">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESMZDilhrgjsDh3MEu">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://appautomation.lambdatest.com/test?testID=APPTESEDy5lm6rcK15ZI5n">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/src/status_im/contexts/wallet/send/events.cljs
+++ b/src/status_im/contexts/wallet/send/events.cljs
@@ -883,23 +883,35 @@
  (fn [{db :db} [value]]
    {:db (assoc-in db [:wallet :ui :send :tx-settings :nonce :current] value)}))
 
-(rf/reg-event-fx :wallet/quick-fee-mode-confirmed
- (fn [{db :db} [fee-mode]]
-   (let [gas-rate         (transaction-settings/tx-fee-mode->gas-rate fee-mode)
-         route            (first (get-in db [:wallet :ui :send :route]))
-         path-tx-identity (send-utils/path-identity route)
-         params           [path-tx-identity gas-rate]]
-     {:db (assoc-in db [:wallet :ui :send :custom-tx-settings :tx-fee-mode] fee-mode)
-      :fx [[:json-rpc/call
-            [{:method   "wallet_setFeeMode"
-              :params   params
-              :on-error (fn [error]
-                          (log/error "failed to set quick transaction settings"
-                                     {:event  :wallet/quick-fee-mode-confirmed
-                                      :error  (:message error)
-                                      :params params}))}]]
-           [:dispatch [:wallet/mark-user-tx-settings-for-deletion]]]})))
+(defn set-fee-mode-effect
+  [path-tx-identity gas-rate]
+  (let [params [path-tx-identity gas-rate]]
+    [:json-rpc/call
+     [{:method   "wallet_setFeeMode"
+       :params   params
+       :on-error (fn [error]
+                   (log/error "failed to set quick transaction settings"
+                              {:event  :wallet/quick-fee-mode-confirmed
+                               :error  (:message error)
+                               :params params}))}]]))
 
+(rf/reg-event-fx
+ :wallet/quick-fee-mode-confirmed
+ (fn [{db :db} [fee-mode]]
+   (let [gas-rate        (transaction-settings/tx-fee-mode->gas-rate fee-mode)
+         tx-type         (get-in db [:wallet :ui :send :tx-type])
+         route           (first (get-in db [:wallet :ui :send :route]))
+         ;; bridge consist from 2 transactions - approval and send, so we need to apply
+         ;; setting to both of them by making 2 calls
+         set-fee-effects (if (= tx-type :tx/bridge)
+                           [(set-fee-mode-effect (send-utils/path-identity route true) gas-rate)
+                            (set-fee-mode-effect (send-utils/path-identity route false) gas-rate)]
+                           [(set-fee-mode-effect (send-utils/path-identity route) gas-rate)])]
+     {:db (assoc-in db [:wallet :ui :send :custom-tx-settings :tx-fee-mode] fee-mode)
+      :fx (conj set-fee-effects
+                [:dispatch [:wallet/mark-user-tx-settings-for-deletion]])})))
+(comment
+  (send-utils/path-identity {:bridge-name "hello"} false))
 
 ;; There is a delay between the moment when user selected
 ;; custom settings and the moment when new route arrived

--- a/src/status_im/contexts/wallet/send/events.cljs
+++ b/src/status_im/contexts/wallet/send/events.cljs
@@ -910,8 +910,6 @@
      {:db (assoc-in db [:wallet :ui :send :custom-tx-settings :tx-fee-mode] fee-mode)
       :fx (conj set-fee-effects
                 [:dispatch [:wallet/mark-user-tx-settings-for-deletion]])})))
-(comment
-  (send-utils/path-identity {:bridge-name "hello"} false))
 
 ;; There is a delay between the moment when user selected
 ;; custom settings and the moment when new route arrived

--- a/src/status_im/contexts/wallet/send/utils.cljs
+++ b/src/status_im/contexts/wallet/send/utils.cljs
@@ -241,9 +241,11 @@
   (not (constants/bridge-assets token-symbol)))
 
 (defn path-identity
-  [path]
-  {:routerInputParamsUuid (:router-input-params-uuid path)
-   :pathName              (:bridge-name path)
-   :chainID               (get-in path [:from :chain-id])
-   :isApprovalTx          (:approval-required path)
-   :communityID           nil})
+  ([path]
+   (path-identity path (:approval-required path)))
+  ([path is-approval-tx?]
+   {:routerInputParamsUuid (:router-input-params-uuid path)
+    :pathName              (:bridge-name path)
+    :chainID               (get-in path [:from :chain-id])
+    :isApprovalTx          is-approval-tx?
+    :communityID           nil}))

